remotespark/RemoteSparkMagics.py

```diff
@@ -125,40 +123,35 @@ def spark(self, line, cell=""):
         # run
         elif len(subcommand) == 0:
             if args.context == Constants.context_name_spark:
-                (success, out) = self.spark_controller.run_cell(cell,
-                                                                args.session)
+                (success, out) = self.spark_controller.run_cell(cell, args.session)
                 if success:
-                    self.ipython.write(out)
+                    self.shell.write(out)
```

Comment:
Did you verify that `shell` will always point to an `InteractiveShell` instance?  Otherwise this will crash at runtime.

`get_ipython()` never returns `None` from what I can tell so I don't see the reason to move away from it.